design/src/rfcs/rfc0012_crates_io_publishing.md

```diff
@@ -0,0 +1,214 @@
+
+> Status: RFC
+
+During its alpha and dev preview releases, the AWS SDK for Rust adopted [a short-term solution
+for versioning and publishing to crates.io](./rfc0011_crates_io_alpha_publishing.md).
+This doc proposes a long-term versioning and publishing strategy that will carry the SDK
+from dev preview into general availability.
+
+This strategy will be implemented in three phases:
+1. __New versioning strategy__: The SDK will break with its current version strategy
+   of maintaining consistent `major.minor` version numbers. This change will be made
+   in isolation for a period of time so that kinks can be worked out with it.
+2. __Full release automation__: Once the new versioning scheme is stable, the release
+   process will be fully automated. Developers will only need to kick off releases manually
+   for changes made to [smithy-rs].
+3. __Stability and 1.x__: This phase begins when the SDK becomes generally available. The
+   major version will be bumped to 1, and backwards breaking changes will no longer be allowed
+   without a major version bump to all crates in the SDK.
+
+Terminology
+-----------
+
+- **AWS SDK Crate**: A crate that provides a client for calling a given AWS service, such as `aws-sdk-s3` for calling S3.
+- **AWS Runtime Crate**: Any runtime crate that the AWS SDK generated code relies on, such as `aws-types`.
+- **Smithy Runtime Crate**: Any runtime crate that the [smithy-rs] generated code relies on, such as `smithy-types`.
+
+Requirements
+------------
+
+### Versioning
+
+Cargo uses [semver] for versioning, with a `major.minor.patch-pre` format:
+- `major`: Incompatible API changes
+- `minor`: Added functionality in backwards compatible manner
+- `patch`: Backwards compatible bug fixes
+- `pre`: Pre-release version tag (omitted for normal releases)
+
+In the new versioning strategy, the `minor` version number will no longer be coordinated across
+all SDK and Smithy runtime crates.
+
+During phases 1 and 2, the `major` version will always be 0, and the following scheme will be used:
+- `minor`:
+  - New features
+  - Breaking changes
+  - Dependency updates for dependencies that are part of the public API
+  - Model updates with API changes
+  - For code-generated crates: when a newer version of [smithy-rs] is used to generate the crate
+- `patch`:
+  - Bug fixes that do not break backwards compatibility
+  - Model updates that _only_ have documentation changes
+
+During phase 3:
+- `major`: Breaking changes
+- `minor`:
+    - Changes that aren't breaking
+    - Dependency updates for dependencies that are part of the public API
+    - Model updates with API changes
+    - For code-generated crates: when a newer version of [smithy-rs] is used to generate the crate
+- `patch`:
+    - Bug fixes that do not break backwards compatibility
+    - Model updates that _only_ have documentation changes
+
+During phase 3, bumps to the `major` version must be coordinated across all SDK and runtime crates.
```

Comment:
Another way of saying this is that "if we break one SDK/runtime crate, we break them all" right?

Reply:
Yeah, essentially.